### ISSUE TITLE
Regression with identifier already declared?

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#8384]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 7.7.4
* **npm Version:** 4.1.2

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```
{
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
const A = {};
Object.defineProperty(A, 'b', {
    get: function () {},
    set: function () {
        const A = function A () {

        };
        A();
    }
});

```

**What did you expect to happen?**

Not to get any errors (as with version 3.18.0).

**What actually happened? Please include the actual, raw output from ESLint.**

```
5:28  error  Parsing error: Identifier 'A' has already been declared
```

If removing the `A` after `function`, it no longer errs. (And yes, I know this is a meaningless example, but just boiled it down.)

I don't have a full grasp of all of the scoping rules, but it would seem to me that the inner function name  ought not be of relevance here.

**Comment by mysticatea** [comment]
Thank you for this report.

This is an upstream's bug (https://github.com/ternjs/acorn/issues/534) which should have been fixed already.
Please try re-install eslint.

Closing as duplicated to #8378.
